lib/dependencies/HarmonyTopLevelThisParserPlugin.js

```diff
@@ -0,0 +1,26 @@
+/*
+	MIT License http://www.opensource.org/licenses/mit-license.php
+	Author Florent Cailhol @ooflorent
+*/
+"use strict";
+
+const ConstDependency = require("./ConstDependency");
+
+class HarmonyTopLevelThisParserPlugin {
+	apply(parser) {
+		let isHarmony = false;
+		parser.hooks.program.tap("HarmonyTopLevelThisParserPlugin", () => {
```

Comment:
This hook is not needed. You can move the isHarmony logic into the `this expression` hook